MigrationExecutorApp/DocumentFeedObserver.cs

```diff
@@ -162,5 +169,50 @@ private static void LogMetrics(IChangeFeedObserverContext context, BulkImportRes
                 //    + ". The failed import docs are: " + failedImportDocs);
             }
         }
+
+        public static Document MapPartitionKey(Document doc, Boolean isSyntheticKey, string targetPartitionKey, Boolean isNestedAttribute, string sourcePartitionKeys)
+        {            
+            if (isSyntheticKey)
+            {
+                doc = CreateSyntheticKey(doc, sourcePartitionKeys, isNestedAttribute, targetPartitionKey);
+            }
+            else
+            {
+                doc.SetPropertyValue(targetPartitionKey, isNestedAttribute == true ? GetNestedValue(doc, sourcePartitionKeys): doc.GetPropertyValue<string>(sourcePartitionKeys));      
+            }
+            return doc;
+        }
+
+        public static Document CreateSyntheticKey(Document doc, string sourcePartitionKeys, Boolean isNestedAttribute, string targetPartitionKey)
+        {
+            StringBuilder syntheticKey = new StringBuilder();
+            string[] sourceAttributeArray = sourcePartitionKeys.Split(',');
+            int arraylength = sourceAttributeArray.Length;
+            int count = 1;
+            foreach (string attribute in sourceAttributeArray)
```

Comment:
Trim the attributes so as to remove white spaces from user input such as "deviceId, timestamp "?